aptos/api/aux-ts/src/graphql/resolvers/mutation.ts

```diff
@@ -268,6 +278,161 @@ export const mutation = {
         .toString(),
     });
   },
+
+  async createStakePool(
+    _parent: any,
+    {
+      createStakePoolInput: { stakePoolInput, rewardAmount, durationUs },
+    }: MutationCreateStakePoolArgs
+  ) {
+    let coinInfoReward = await auxClient.getCoinInfo(
+      stakePoolInput.coinTypeReward
+    );
+    let coinInfoStake = await auxClient.getCoinInfo(
+      stakePoolInput.coinTypeStake
+    );
+    const stakePoolClient = new StakePoolClient(auxClient, {
+      coinInfoReward,
+      coinInfoStake,
+    });
+    return stakePoolClient.create({
+      rewardAmount: DU(rewardAmount),
+      durationUs: new BN(durationUs),
+    });
```

Comment:
this will actually fail because there won't be a `AptosAccount` to sign. what the other mutation resolvers actually send back is just the object representing the move call (`Types.UserTransaction`), and that corresponding object has to be invoked by the wallet

i'm sorry for the confusion. mutation is to namespace user writes from reads

Reply:
Ah ok, I think I was just going off https://github.com/aux-exchange/aux-exchange/blob/937490e37b97b61bff2f67e7004b9444fddf16f4/aptos/api/aux-ts/src/graphql/resolvers/mutation.ts#L60-L63 and didn't look closely at the other methods. Does that mean that `createPool` also needs to be fixed to do the same?

Reply:
yeah

Reply:
fixed